### PR TITLE
feature(incidence-reports): fetch all incidence reports

### PR DESCRIPTION
**Pull request by kimotho-njoki**
## What does this PR do?
This PR fetches all incidence reports from the API and makes it possible to view all reports using the link option on the Side Menu.

## Description of Task to be completed?
- Use the ART Backend API endpoint to show all incidence reports
- AS a USER
   WHEN I click on the incidence report link on the sidemenu
    - It should fetch all incident reports from the backend and display paginated results in a table
  
## How should this be manually tested?
- Pull the branch and click on `reports` on the Side Menu.

## What are the relevant pivotal tracker stories?
[158492395](https://www.pivotaltracker.com/n/projects/2146417/stories/158492395)

## Any background context you want to add?
N/A 

## Important notes
N/A

## Packages installed
N/A

## Deployment note
N/A

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation
## Screenshots
screen shot 2018-06-27 at 21 02 55

### CODE REVIEW

File: src/__test__/actions/incidenceReports.actions.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// third-party libraries
+import expect from 'expect';
+import MockAdapter from 'axios-mock-adapter';
+import axios from 'axios';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+
+// constants
+import constants from '../../_constants';
+const {
```
Comment (sachungo) [inline src/__test__/actions/incidenceReports.actions.test.js]: This should be after all imports. Therefore it should be declared after mock data import

File: src/__test__/reducers/incidenceReports.reducer.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// actions
+import {
+  loadIncidenceReportsSuccess,
+  loadIncidenceReports
+} from '../../_actions/incidenceReports.actions';
+
+describe('Incidence Reports Reducer', () => {  
+  it('should handle LOAD_INCIDENCE_REPORTS_SUCCESS', () => {
+    let expected = {
+        reports: incidenceReports,
```
Comment (sachungo) [inline src/__test__/reducers/incidenceReports.reducer.test.js]: The indentation is off. Please use 2 spaces. I.e
```
const expected = {
  reports: incidenceReports,
  ...
};
```

File: src/__test__/reducers/incidenceReports.reducer.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+// actions
+import {
+  loadIncidenceReportsSuccess,
+  loadIncidenceReports
+} from '../../_actions/incidenceReports.actions';
+
+describe('Incidence Reports Reducer', () => {  
+  it('should handle LOAD_INCIDENCE_REPORTS_SUCCESS', () => {
+    let expected = {
```
Comment (sachungo) [inline src/__test__/reducers/incidenceReports.reducer.test.js]: use `const` as there's no reassignment done to the variable later on

File: src/__test__/reducers/incidenceReports.reducer.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        reports: incidenceReports,
+        incidenceReportsCount: 2,
+        isLoading: false,
+        hasError: false,
+    };
+
+    let action = loadIncidenceReportsSuccess(incidenceReports);
```
Comment (sachungo) [inline src/__test__/reducers/incidenceReports.reducer.test.js]: use `const`

File: src/_actions/incidenceReports.actions.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * Incidence reports thunk
+ *
+ * @return {(dispatch:any)=>Promise<TResult2|TResult1>}
+ */
+export const loadIncidenceReports = (page = 1, limit = 10) =>
+  (dispatch) => {
+    dispatch({ type: LOAD_INCIDENCE_REPORTS_START });
+    return axios.get('http://art-backend.herokuapp.com/api/v1/incidence-reports/')
```
Comment (sachungo) [inline src/_actions/incidenceReports.actions.js]: This should be `axios.get('incidence-reports')`. The API URL is supposed to be declared in your `.env` as `REACT_APP_API_URL`. Read [this slack message](https://andela.slack.com/archives/GAMEXMPFV/p1528813884000969)

File: src/components/IncidenceReportsComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  handlePaginationChange = (e, { activePage }) => {
+    this.setState({ activePage });
+    this.props.loadIncidenceReports(activePage);
+  }
+
+  handlePageTotal = () => Math.ceil(this.props.incidenceReportsCount / this.state.limit)
+
+  emptyReportsCheck = () => (this.props.reports.length === 0)
```
Comment (sachungo) [inline src/components/IncidenceReportsComponent.jsx]: Please use lodash's `isEmpty` method

File: src/components/IncidenceReportsComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+            {
+              this.loadTableContent()
+            }
+          </Table.Body>
+
+          <Table.Footer>
+            <Table.Row>
+              <Table.HeaderCell colSpan="8">
+                {
+                  (this.emptyReportsCheck())
```
Comment (sachungo) [inline src/components/IncidenceReportsComponent.jsx]: A ternary operator whose one expression renders nothing to the UI can be simplified with `&&` to:
```
{!this.emptyReportsCheck() && (
  <Pagination
    totalPages={this.handlePageTotal()}
    onPageChange={this.handlePaginationChange}
    activePage={this.state.activePage}
  />
)}
```

File: src/components/IncidenceReportsComponent.jsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  handlePaginationChange = (e, { activePage }) => {
+    this.setState({ activePage });
+    this.props.loadIncidenceReports(activePage);
+  }
+
+  handlePageTotal = () => Math.ceil(this.props.incidenceReportsCount / this.state.limit)
+
+  emptyReportsCheck = () => (this.props.reports.length === 0)
+
+  loadTableContent = () => {
```
Comment (sachungo) [inline src/components/IncidenceReportsComponent.jsx]: I usually discourage the use of a class method to separate the markup. This tends to build up overtime, making a file too long if future markup additions is done using the same pattern. If you find yourself doing this, then it's time to consider if you need a new component or you can use a ternary operator. Note that if you use a ternary operator and the code looks unreadable, then it means there's a better way of doing it. If in doubt, please come and ask me for clarification.

File: src/_mock/incidenceReports.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+export default [
+  {
+    id: 0,
+    asset: 'Mac Book',
+    incident_type: 'Loss',
+    incident_location: 'Nairobi',
+    incident_description: 'Was stolen',
+    injuries_sustained: 'Null',
```
[the remark this answers is not in the document]
Comment (kimotho-njoki) [reply src/_mock/incidenceReports.js]: I honestly just ran out of ideas on what to place as a suitable mock string. Let me fix that right away as well as the rest of the comments. @sachungo

**Pull request merged**